### PR TITLE
Allow old school rule-like policies with RulePolicy

### PR DESCRIPTION
**Pull request by wochinge**
**Proposed changes**:
- fix https://github.com/RasaHQ/rasa/issues/6820#issuecomment-702246054
- make it possible to use the `RulePolicy` in conjunction with the old, deprecated "rule-style" policies. Motivation: As a user I want to migrate the assistant step by step. The procedure is roughly the following
   1. Migrate one policy
   2. Train assistant
   3. test
   4. Migrate next policy
  This was not possible without this PR as the training would have thrown an exception about the combined usage. This exception is now relaxed to a warning.
- give the `RulePolicy` a higher priority to make sure it can be used in conjunction with `MappingPolicy` and co
- change check for `FormPolicy` in `rasa data convert config` to a warning

**Status (please check what you already did)**:
- [x] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [x] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

**Comment by wochinge** [comment]
I'll merge this directly into master. This is just targeted to `rasa-migrate-config` to keep the diff small.

**Review by m-vdb — COMMENTED** [review]
LGTM 👍  but I think it's worth a review from someone who knows more about the Rule Policy

File: docs/docs/policies.mdx [hunk 1]
```diff
@@ -52,7 +52,7 @@ and Rule Policies might both predict with confidence 1), the priority of the
 policies is considered. Rasa policies have default priorities that are set to ensure the
 expected outcome in the case of a tie. They look like this, where higher numbers have higher priority:
 
-5. `RulePolicy`
+6. `RulePolicy`
```
Comment (Ghostvv) [inline docs/docs/policies.mdx:55]: I think these numbers will be confusing for new 2.0 users who are not aware of old policies

**Review by Ghostvv — APPROVED** [review]

File: docs/docs/policies.mdx (hunk 1 above)
Comment (wochinge) [reply docs/docs/policies.mdx:55]: @erohmensing I think Vova has a point. Why did we remove the deprecated polices from this? I'm gonna merge my change anyway as this outside of the scope this PR imo.

**Comment by wochinge** [comment]
I thought `rasa-migrate-config` would be long merged but it isn't. Merging everything together now to handle only 2 prs instead of 3

**Pull request merged**